src/meta/src/hummock/compaction/tier_compaction_picker.rs

```diff
@@ -196,3 +196,63 @@ impl CompactionPicker for TierCompactionPicker {
         self.pick_whole_level(l0, &level_handlers[0])
     }
 }
+
+#[cfg(test)]
+pub mod tests {
+    use std::sync::Arc;
+
+    use risingwave_pb::hummock::hummock_version::Levels;
+    use risingwave_pb::hummock::LevelType;
+
+    use crate::hummock::compaction::compaction_config::CompactionConfigBuilder;
+    use crate::hummock::compaction::level_selector::tests::{
+        generate_l0_overlapping_sublevels, generate_level, generate_table,
+    };
+    use crate::hummock::compaction::overlap_strategy::RangeOverlapStrategy;
+    use crate::hummock::compaction::{CompactionPicker, TierCompactionPicker};
+    use crate::hummock::level_handler::LevelHandler;
+
+    #[test]
+    fn test_pick_whole_level_skip_sublevel() {
+        let l0 = generate_l0_overlapping_sublevels(vec![
+            vec![
+                generate_table(4, 1, 10, 90, 1),
+                generate_table(5, 1, 210, 220, 1),
+            ],
+            vec![generate_table(6, 1, 0, 100, 1)],
+            vec![generate_table(7, 1, 0, 100, 1)],
+        ]);
+        let mut levels = Levels {
+            l0: Some(l0),
+            // This L1 prevents any trivial move.
+            levels: vec![generate_level(1, vec![generate_table(3, 1, 0, 100000, 1)])],
+        };
+        levels.l0.as_mut().unwrap().sub_levels[0].level_type = LevelType::Nonoverlapping as i32;
```

Comment:
Accoording to our compaction strategy, is it possible to have a nonoverlapping sublevel followed by overlapping sublevels? I thought an overlapping sublevel can only exist in the prefix of our sublevels.

Reply:
cc @Little-Wallace

Reply:
> is it possible to have a nonoverlapping sublevel followed by overlapping sublevels

I think so. New sublevel must be overlapping and be added in the end of sublevels.
Please confirm @Little-Wallace

Reply:
The strategy can guarantee that we would not select some overlapping skip other overlapping level.
But if the previous compact task which select some overlapping level failed, it would cause a nonoverlapping sublevel followed by overlapping level.

Reply:
> But if the previous compact task which select some overlapping level failed, it would cause a nonoverlapping sublevel followed by overlapping level.

I guess you mean "cause a nonoverlapping followed overlapping"?
So nonoverlapping and overlapping sublevels can interleave in any order in L0, right?